aldryn_forms/cms_plugins.py

```diff
@@ -494,6 +496,39 @@ def form_pre_save(self, instance, form, request, **kwargs):
         form.cleaned_data[field_name] = filer_file
 
 
+class ImageField(FileField):
+    name = _('Image upload field')
+
+    model = models.ImageUploadFieldPlugin
+
+    form = ImageFieldForm
+    form_field = RestrictedImageField
+    form_field_widget = RestrictedImageField.widget
+    fieldset_general_fields = Field.fieldset_general_fields + [
+        'upload_to',
```

Comment:
Files have a random component in the url anyway, so it is already virtually impossible to access the files without the full url.

For the files to be password protected, the project has to configure django-filer to use a secure storage backend and serve the files through django (http://django-filer.readthedocs.org/en/latest/secure_downloads.html). Also permissions have to be set on the folder.